src/benchmarklib/benchmark_utils.cpp

```diff
@@ -31,28 +31,41 @@ std::ostream& get_out_stream(const bool verbose) {
 }
 
 BenchmarkState::BenchmarkState(const size_t max_num_iterations, const opossum::Duration max_duration)
-    : max_num_iterations(max_num_iterations), max_duration(max_duration) {}
+    : max_num_iterations(max_num_iterations), max_duration(max_duration) {
+  iteration_durations.reserve(max_num_iterations);
+}
 
 bool BenchmarkState::keep_running() {
+  bool is_first_iteration = false;
+
   switch (state) {
     case State::NotStarted:
-      begin = std::chrono::high_resolution_clock::now();
+      benchmark_begin = std::chrono::high_resolution_clock::now();
       state = State::Running;
+      is_first_iteration = true;
       break;
     case State::Over:
       return false;
     default: {}
   }
 
+  const auto now = std::chrono::high_resolution_clock::now();
```

Comment:
`iteration_start`?

Reply:
I thought about this as well, but this makes the following code harder to read:
```
const auto iteration_duration = iteration_start - iteration_begin;
```
and
```
benchmark_end = iteration_start;
```
seem really confusing to me.

Reply:
ok